lua/lspconfig/leanls.lua

```diff
@@ -3,21 +3,25 @@ local util = require 'lspconfig/util'
 
 configs.leanls = {
   default_config = {
-    cmd = {"lean-language-server", "--stdio"};
+    cmd = {"lean", "--server"};
     filetypes = {"lean"};
     root_dir = function(fname)
-      return util.find_git_ancestor(fname) or vim.loop.os_homedir()
+      return util.root_pattern("leanpkg.toml")(fname) or util.find_git_ancestor(fname) or util.path.dirname(fname)
+    end;
+    on_new_config = function(config, root)
+      if not util.path.is_file(root .. "/leanpkg.toml") then return end
+      if not config.cmd_cwd then
+        config.cmd_cwd = root
+      end
     end;
   };
   docs = {
-    package_json = "https://raw.githubusercontent.com/leanprover/vscode-lean/master/package.json";
     description = [[
-https://github.com/leanprover/lean-client-js/tree/master/lean-language-server
-
-Lean language server.
+The Lean 4 language server is built-in with a Lean 4 install,
+and can be run with, e.g., "lean --server".
     ]];
     default_config = {
-      root_dir = [[root_pattern(".git") or os_homedir]];
+      root_dir = [[find_toml() or root_pattern(".git") or os_homedir]];
```

Comment:
generally don't make of these function names, find_toml isn't a real function neither is os_homedir.

Reply:
Whoops, forgot to update this. Thanks!